common/dynamicconfig/constants.go

```diff
@@ -1056,7 +1056,12 @@ See DynamicRateLimitingParams comments for more details.`,
 	MatchingGetTasksBatchSize = NewTaskQueueIntSetting(
 		"matching.getTasksBatchSize",
 		1000,
-		`MatchingGetTasksBatchSize is the maximum batch size to fetch from the task buffer`,
+		`How many backlog tasks to read from persistence at once`,
+	)
+	MatchingGetTasksReloadAt = NewTaskQueueIntSetting(
+		"matching.getTasksReloadAt",
+		100,
+		`Reload a batch of tasks when there are this many remaining. Must be less than MatchingGetTasksBatchSize.`,
```

Comment:
Is my understanding correct that if `MatchingGetTasksBatchSize + MatchingGetTasksReloadAt >= MatchingGetTasksBatchSize` we would never load tasks from the backlog? I'm wondering if there's a trapdoor here for operators (like ourselves). Since this can be changed on the fly, a hard fail on startup isn't an option. But we could consider capping the value to `MatchingGetTasksBatchSize`?

Reply:
Did you miss a 2× in that comparison?

Yeah, it's possible to set this so that it breaks things. But that's possible for a lot of dynamic configs. I'd like to create a general validation mechanism eventually, but that'll need some design work.

For this one, I'm thinking of it as almost temporary, ideally we'd have some code that tracks the dispatch rate and reloads based on time left (e.g. 1s before we're about to run out) rather than tasks left.

Reply:
Then we'd have to read two configs instead of one each time we check it?  I think the logic can be improved (e.g. if the config is changed it might miss the trigger), but I'd like to do it in another PR